src/Aspire.Hosting.Testing/DistributedApplicationFactory.cs

```diff
@@ -131,16 +134,66 @@ private void OnBuilderCreatingCore(DistributedApplicationOptions applicationOpti
         applicationOptions.AssemblyName = _entryPoint.Assembly.GetName().Name ?? string.Empty;
         applicationOptions.DisableDashboard = true;
         var cfg = hostBuilderOptions.Configuration ??= new();
-        cfg.AddInMemoryCollection(new Dictionary<string, string?>
+        var additionalConfig = new Dictionary<string, string?>
         {
             ["DcpPublisher:RandomizePorts"] = "true",
             ["DcpPublisher:DeleteResourcesOnShutdown"] = "true",
             ["DcpPublisher:ResourceNameSuffix"] = $"{Random.Shared.Next():x}",
-        });
+        };
+
+        var appHostProjectPath = ResolveProjectPath(_entryPoint.Assembly);
+        if (!string.IsNullOrEmpty(appHostProjectPath))
+        {
+            hostBuilderOptions.ContentRootPath = appHostProjectPath;
+        }
+
+        var appHostLaunchSettings = GetLaunchSettings(appHostProjectPath);
+        if (appHostLaunchSettings?.Profiles.FirstOrDefault().Key is { } profileName)
+        {
+            additionalConfig["AppHost:DefaultLaunchProfileName"] = profileName;
+        }
+
+        cfg.AddInMemoryCollection(additionalConfig);
 
         OnBuilderCreating(applicationOptions, hostBuilderOptions);
     }
 
+    private static string? ResolveProjectPath(Assembly? assembly)
+    {
+        var assemblyMetadata = assembly?.GetCustomAttributes<AssemblyMetadataAttribute>();
+        return GetMetadataValue(assemblyMetadata, "AppHostProjectPath");
+    }
+
+    private static string? GetMetadataValue(IEnumerable<AssemblyMetadataAttribute>? assemblyMetadata, string key)
+    {
+        return assemblyMetadata?.FirstOrDefault(m => string.Equals(m.Key, key, StringComparison.OrdinalIgnoreCase))?.Value;
+    }
+
+    private static LaunchSettings? GetLaunchSettings(string? appHostPath)
+    {
+        if (appHostPath is null || !Directory.Exists(appHostPath))
+        {
+            return null;
+        }
+
+        var projectFileInfo = new DirectoryInfo(appHostPath);
+        var launchSettingsFilePath = projectFileInfo.FullName switch
+        {
+            null => Path.Combine("Properties", "launchSettings.json"),
```

Comment:
We might want to check if casing for the path is enforced in other places that use it, e.g. `dotnet run`. Would be a pain if `properties/launchsettings.json` was read just fine by `dotnet run` on Linux, but not here.

Reply:
`dotnet run` uses the same logic, although they have a case for VB which I think we're ok to ignore: https://github.com/dotnet/sdk/blob/2581c9040605e81811d5d7b4257684c04f1fb382/src/Cli/dotnet/commands/dotnet-run/RunCommand.cs#L135-L140